lib/trie/database.go

```diff
@@ -97,7 +97,7 @@ func (t *Trie) loadFromProof(rawProof [][]byte, rootHash []byte) error {
 		decodedNode.SetDirty(dirty)
 		decodedNode.SetEncodingAndHash(rawNode, nil)
 
-		_, hash, err := decodedNode.EncodeAndHash()
+		_, hash, err := decodedNode.EncodeAndHash(decodedNode == t.root)
```

Comment:
`decodedNode == t.root` is always false here (which is fine, since it's the same functionality as before)

Reply:
but perhaps just replace the case with `false`?